cumulus/parachains/runtimes/testing/penpal/src/xcm_config.rs

```diff
@@ -318,8 +325,62 @@ pub type Reserves = (
 	NativeAssetFrom<SystemAssetHubLocation>,
 	AssetPrefixFrom<EthereumLocation, SystemAssetHubLocation>,
 );
-pub type TrustedTeleporters =
-	(AssetFromChain<LocalTeleportableToAssetHub, SystemAssetHubLocation>,);
+
+pub struct ConcreteAssetFromBridgeHub<AssetLocation>(PhantomData<AssetLocation>);
+impl<AssetLocation: Get<Location>> ContainsPair<Asset, Location>
+	for ConcreteAssetFromBridgeHub<AssetLocation>
+{
+	fn contains(asset: &Asset, origin: &Location) -> bool {
+		log::trace!(target: "xcm::contains", "ConcreteAssetFromBridgeHub asset: {:?}, origin: {:?}", asset, origin);
+		let is_from_bh = match origin.unpack() {
+			// System parachain
+			(1, [Parachain(id)]) => *id == 1013,
+			// Others
+			_ => false,
+		};
+		asset.id.0 == AssetLocation::get() && is_from_bh
+	}
+}
+
+pub type TrustedTeleporters = (
+	ConcreteAssetFromBridgeHub<RelayLocation>,
+	AssetFromChain<LocalTeleportableToAssetHub, SystemAssetHubLocation>,
+);
+
+parameter_types! {
+	pub SiblingBridgeHubLocation: Location = Location::new(
+				1,
+				[
+					Parachain(1013),
+				]
+			);
+	pub UniversalAliases: BTreeSet<(Location, Junction)> = BTreeSet::from_iter(
+			sp_std::vec![
+				(SiblingBridgeHubLocation::get(),GlobalConsensus(EthereumNetwork::get())),
+			]
+	);
+}
+
+pub struct GlobalConsensusEthereumAddressConvertsFor<AccountId>(PhantomData<AccountId>);
+impl<AccountId> ConvertLocation<AccountId> for GlobalConsensusEthereumAddressConvertsFor<AccountId>
+where
+	AccountId: From<[u8; 32]> + Clone,
+{
+	fn convert_location(location: &Location) -> Option<AccountId> {
+		match location.unpack() {
+			(_, [first_loc, AccountKey20 { network: None, key: sender }])
+				if *first_loc == Junction::from(EthereumNetwork::get()) =>
+				Some(blake2_256(&(b"AccountKey20", sender).encode()).into()),
```

Comment:
Should we include chain_id in the hash?

Reply:
I would prefer to not include it then the same Ethereum account will always map to the same substrate account(i.e. same address for mainnet and sepolia).